atlasdb-config/src/main/java/com/palantir/atlasdb/factory/TransactionManagers.java

```diff
@@ -518,17 +519,36 @@ static LockAndTimestampServices createRawInstrumentedServices(
             com.google.common.base.Supplier<TimestampService> time,
             TimestampStoreInvalidator invalidator,
             String userAgent) {
+        AtlasDbRuntimeConfig initialRuntimeConfig = runtimeConfigSupplier.get();
+        assertNoSpuriousTimeLockBlockInRuntimeConfig(config, initialRuntimeConfig);
         if (config.leader().isPresent()) {
             return createRawLeaderServices(config.leader().get(), env, lock, time, userAgent);
         } else if (config.timestamp().isPresent() && config.lock().isPresent()) {
             return createRawRemoteServices(config, userAgent);
-        } else if (config.timelock().isPresent()) {
+        } else if (config.timelock().isPresent() || initialRuntimeConfig.timelockRuntime().isPresent()) {
             return createRawServicesFromTimeLock(config, runtimeConfigSupplier, invalidator, userAgent);
         } else {
             return createRawEmbeddedServices(env, lock, time);
         }
     }
 
+    private static void assertNoSpuriousTimeLockBlockInRuntimeConfig(
+            AtlasDbConfig config,
+            AtlasDbRuntimeConfig initialRuntimeConfig) {
+        // Note: The other direction (timelock install config without a runtime block) should be maintained for
+        // backwards compatibility.
+        if (timestampOrLockOrLeaderBlockPresent(config) && initialRuntimeConfig.timelockRuntime().isPresent()) {
+            throw new IllegalStateException("Found a service configured not to use timelock, with a timelock"
+                    + " block in the runtime config! This is unexpected. If you wish to use non-timelock services,"
+                    + " please remove the timelock block from the runtime config; if you wish to use timelock,"
+                    + " please remove the leader, remote timestamp or remote lock configuration blocks.");
+        }
+    }
+
+    private static boolean timestampOrLockOrLeaderBlockPresent(AtlasDbConfig config) {
+        return config.timestamp().isPresent() || config.lock().isPresent() || config.leader().isPresent();
```

Comment:
nit: we should either check for only `timestamp` or `timestamp` && `lock`. This kinda conflicts with the top level config check.

Reply:
Fair enough. I'd rather not check _only_ tor timestamp - think `timestamp && lock` is clearer in terms of intention, so will switch to that.